### PR TITLE
New file install mode: symlink+

### LINKED ISSUE #1893: rose file install: check target of symlink
Reported by a user here - configuration that looks like this:

``` ini
[file:myfile]
source=/non/existing/path
mode=symlink
```

will happily produce a broken symlink without any warnings or errors.

### PR DESCRIPTION
**Pull request by matthewrmshin**
Check symlink target exists before creating symlink.

Close #1893. @arjclark @oliver-sanders please review.

### CODE REVIEW

File: doc/rose-rug-files.html [hunk 1]
```diff
@@ -427,11 +427,21 @@ <h3 class="alwayshidden">Symbolic Links</h3>
 
                 <p>If this file is a bit large to copy over, you
                 could consider asking for a symbolic link instead
-                of a copy:</p>
+                of a copy. Use <code>mode=symlink+</code> to
+                ensure that the target of the symbolic link exists.
+                Use <code>mode=symlink</code> if the target of the
+                symbolic link is not guarantee to exist at the
```
Comment (arjclark) [inline doc/rose-rug-files.html]: guarantee*d*

File: doc/rose-rug-files.html [hunk 2]
```diff
@@ -427,11 +427,21 @@ <h3 class="alwayshidden">Symbolic Links</h3>
...
+                time the symboloc link is created. (E.g. the
```
Comment (arjclark) [inline doc/rose-rug-files.html]: symbol*ic*

**Comment by arjclark** [comment]
All yours @oliver-sanders

File: t/rose-app-run/25-file-symlink-plus.t [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+test_setup
+run_fail "${TEST_KEY}" rose app-run --config=../config -q
+file_cmp "${TEST_KEY}.out" "${TEST_KEY}.out" </dev/null
+file_cmp "${TEST_KEY}.err" "${TEST_KEY}.err" <<'__ERR__'
+[FAIL] file:COPYING=source=fantasy.txt: [Errno 2] No such file or directory: 'fantasy.txt'
+__ERR__
+run_fail "${TEST_KEY}-test-l" test -L 'COPYING'
+test_teardown
+#-------------------------------------------------------------------------------
+TEST_KEY="${TEST_KEY_BASE}"
```
Comment (oliver-sanders) [inline t/rose-app-run/25-file-symlink-plus.t]: Minor comment, these `TEST_KEY`'s are the same.

Comment (matthewrmshin) [inline t/rose-app-run/25-file-symlink-plus.t]: Updated.

**Pull request merged**